### ISSUE TITLE
error creating empty MVector

### ISSUE THREAD

**Issue by stevengj** [JuliaArrays/StaticArrays.jl#557]
This seems like it should work:
```jl
julia> MVector(SVector{0,Float64}())
ERROR: UndefVarError: T not defined
Stacktrace:
 [1] MArray{Tuple{0},T,1,0} where T(::Tuple{}) at /Users/stevenj/.julia/packages/StaticArrays/3ENSR/src/MVector.jl:20
 [2] Type at /Users/stevenj/.julia/packages/StaticArrays/3ENSR/src/MVector.jl:19 [inlined]
 [3] MArray{Tuple{S},T,1,S} where T where S(::SArray{Tuple{0},Float64,1,0}) at /Users/stevenj/.julia/packages/StaticArrays/3ENSR/src/convert.jl:5
 [4] top-level scope at none:0
```

**Comment by andyferris** [comment]
That does look like a bug. 

In general, we are having trouble with the semantic of one argument constructors - does the user want to convert an array or nest it?

Finally - I’m curious, what do you do with an `SVector{0}` in practice?

**Comment by stevengj** [comment]
I encountered this for generic functionality in HCubature.jl: it has an n-dimensional `SVector` that it converts to an `MVector` for mutation purposes.   The same code still needs to work in the case of n=0, even though the loops are empty in that case.

I worked around this by adding an explicit n=0 check, but it would be nicer if a length-0 `MVector` just worked in order to support edge cases like this.
